### PR TITLE
WT-6677 Map read committed and uncommitted isolation levels to read only

### PR DESCRIPTION
**Pull request by raviprakashgiri29**

### CODE REVIEW

File: src/include/txn_inline.h [hunk 1]
```diff
@@ -1194,6 +1194,15 @@ __wt_txn_id_check(WT_SESSION_IMPL *session)
     if (F_ISSET(txn, WT_TXN_HAS_ID))
         return (0);
 
+    /*
+     * Return error when the transactions with read committed or uncommitted isolation tries to
+     * perform any write operation.
+     */
+    if (session->dhandle != NULL && !WT_IS_METADATA(session->dhandle) &&
```
Comment (sulabhM) [inline src/include/txn_inline.h:1201]: Alex has mentioned as a comment on WT-6677 a few extra things that can be done. Adding a comment around metadata exception and adding an assertion for internal threads. I will wait for those before I review again.

Comment (raviprakashgiri29) [reply src/include/txn_inline.h:1203]: @sulabhM Updated the comments and added an assert for internal threads.

File: src/include/txn_inline.h [hunk 2]
```diff
@@ -1194,6 +1194,19 @@ __wt_txn_id_check(WT_SESSION_IMPL *session)
...
+     * Return error when the transactions with read committed or uncommitted isolation tries to
+     * perform any write operation. Skip metadata because metadata searches (like schema operations)
+     * rely on WT_ISO_READ_UNCOMMITTED and metadata checkpoints rely on WT_ISO_READ_COMMITTED
+     * isolation.
+     */
+    if (session->dhandle != NULL && !WT_IS_METADATA(session->dhandle) &&
+      (txn->isolation == WT_ISO_READ_COMMITTED || txn->isolation == WT_ISO_READ_UNCOMMITTED)) {
+        WT_ASSERT(session, !F_ISSET(session, WT_SESSION_INTERNAL));
+        WT_RET_MSG(
+          session, ENOTSUP, "not supported in read-committed or read-uncommitted transactions.");
```
Comment (kommiharibabu) [inline src/include/txn_inline.h:1207]: I feel the error message should be more clear as `write operations are not ...`?

File: src/include/txn_inline.h [hunk 3]
```diff
@@ -1194,6 +1194,19 @@ __wt_txn_id_check(WT_SESSION_IMPL *session)
     if (F_ISSET(txn, WT_TXN_HAS_ID))
         return (0);
 
+    /*
+     * Return error when the transactions with read committed or uncommitted isolation tries to
+     * perform any write operation. Skip metadata because metadata searches (like schema operations)
```
Comment (kommiharibabu) [inline src/include/txn_inline.h:1199]: The comments related to metadata may be enhanced to get more clarity. How about something like the following?
`Don't raise an error for an update on metadata as the search and updates on metadata happens in read-uncommitted and read-committed isolations compared to the other tables`.

File: src/include/txn_inline.h (hunk 2 above)
Comment (raviprakashgiri29) [reply src/include/txn_inline.h]: Yeah, this sentence is more relevant. Updated the comments

File: src/include/txn_inline.h (hunk 3 above)
Comment (raviprakashgiri29) [reply src/include/txn_inline.h]: Updated comments.

**Comment by raviprakashgiri29** [comment]
The PR testing failure has been recorded in WT-6952.

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Review by sulabhM — APPROVED** [review]

**Pull request merged**